Dev

BenG-SFDC opened the pull request.

BenG-SFDC committed:
top10 pull

added pull for top 10 donators for user

BenG-SFDC committed:
PythonFixes

* added json files to gitignore for good measure
* upated python to hopefully fix utf-8 and encoding bugs on all machines

BenG-SFDC committed:
Scrolling text update

To clarify scrolling text updates

BenG-SFDC committed:
Video Folder setup

Setup Video Folder

BenG-SFDC committed:
video player part two

Updated video player to pull folder items, and play randomly.  Just have to clean it up now

BenG-SFDC committed:
videoplayer - final test

Video player finalization - cleaned up code, added mp4 filter and shifted some variables around

BenG-SFDC committed:
Bump

To Ensure folder creation

BenG-SFDC committed:
Added YAML configuration

 Beginning pulling user variables from individual files to YAML data

BenG-SFDC committed:
YAML config finalization

Finalized moving configs to YAML file, and updating index for more info

BenG-SFDC committed:
DataUpdate as exe fix

Moved EXE file and adjusted configurations to fix small bugs

BenG-SFDC committed:
Batch update and final doc change

this is now ready to be used by someone other than me!

AevumDecessus commented on dataupdate.py:
```diff
@@ -0,0 +1,79 @@
+personID = str(config["pyPersonID"])
+
+# pingSeconds - Number of seconds between server pings
+pingSeconds = config["pyPingSeconds"]
```
You want to validate this and make sure it can never be set to below 15 seconds, for example:
```suggestion
pingSeconds = int(config["pyPingSeconds"])
if pingSeconds < 15:
  pingSeconds = 15
```

BenG-SFDC replied:
In the end, Ill have to change this soon to point to our local copies anyway, but good point.  Changed.

AevumDecessus commented on htdocs/FSSConfig.yml:
```diff
@@ -0,0 +1,56 @@
+---
```
For config files, git best practice is to do the following:
* Add htdocs/FSSConfig.yml to .gitignore
* Copy htdocs/FSSConfig.yml to htdocs/FSSConfig.yml.sample
* Add the .sample config to git

This makes it so your local changes, or an end user's changes to the config file are never overwritten by a git pull

BenG-SFDC replied:
I'm trying to make this runnable with default configs, so leaving the yml file as-is.  That being said, I have added it to .gitignore, so future changes should be untracked.

AevumDecessus replied:
Possible solution is to copy the sample file to the config location on start if the config location doesn't exist? Worth a thought

BenG-SFDC replied:
bumped the yml file to a sample, and threw error handling in the go script

AevumDecessus commented on htdocs/donationalert.html:
```diff
@@ -5,22 +5,33 @@
-		//var jsonURL = '/persondonations.json'; // for personal donations only
-		var aboveLogoString = 'New Team Donation!'; // Text String above logo
+		//load YAML data
+		var yamlConfig = YAML.load('FSSConfig.yml');
```
We need a way to fail gracefully if the file doesn't exist yet (if you follow the git best practices for config files from above)

BenG-SFDC replied:
Not needed per above.

BenG-SFDC committed:
Quick change - code review

BenG-SFDC committed:
quick change - code review

BenG-SFDC committed:
ignoring FSSConfig in git to prevent updates

ravenium committed:
Added Go-based Webserver/poller

BenG-SFDC committed:
Merge pull request #3 from ravenium/dev

Added Go-based Webserver/poller

BenG-SFDC committed:
Remove unnecessary items

BenG-SFDC committed:
further configuration setup

BenG-SFDC committed:
final doc update

BenG-SFDC committed:
renaming YML file to sample file

BenG-SFDC committed:
go changes and re-complie

fixed error to allow for visible failure on double-click and added check for yaml file

BenG-SFDC committed:
depricate videoplayer in docs

BenG-SFDC committed:
Readme Update
